File: leetcode/Remove Duplicates from Sorted Array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut first = true;
        let mut count: usize = 0;
        let mut last: i32 = 0;

        for (pos, e) in nums.clone().iter().enumerate() {
            if *e != last || first {
                last = *e;
            } else {
                nums.remove(pos - count);
                count += 1
            }
            first = false;
        }
        nums.len() as i32
    }
}

File: leetcode/Remove Duplicates from Sorted Array.go
package leetcode

func removeDuplicates(nums []int) int {
	current := 0

	for i := 0; i < len(nums); i++ {
		if nums[current] != nums[i] {
			current++
			nums[current] = nums[i]
		}
	}

	return current + 1
}
